crates/llvm-context/src/polkavm/context/pointer.rs

```diff
@@ -76,6 +76,35 @@ impl<'ctx> Pointer<'ctx> {
         Self::new(r#type, address_space, value)
     }
 
+    /// Creates a new pointer with the specified `offset` for storage operations.
+    pub fn new_with_store_offset<D, T>(
+        context: &Context<'ctx, D>,
+        address_space: AddressSpace,
+        r#type: T,
+        offset: inkwell::values::IntValue<'ctx>,
+        name: &str,
+    ) -> Self
+    where
+        D: Dependency + Clone,
+        T: BasicType<'ctx>,
+    {
+        assert_ne!(
+            address_space,
+            AddressSpace::Stack,
+            "Stack pointers cannot be addressed"
+        );
```

Comment:
This should rather assert that the address space is either Storage or TransientStorage?